### DISCUSSION TITLE
[Kavita] Browse by Author ( & penciller, letterer, editor, etc.?)

### DISCUSSION THREAD

**Discussion by DieselTech** [Kareadita/Kavita#2556] [Ideas]
### Idea Description

Browse by Author ( & penciller, letterer, editor, etc.?)
Posted by JJ_HK
· Feb 2022
· 13 Upvotes
https://feats.kavitareader.com/posts/74/browse-by-author-and-penciller-letterer-editor-etc

Description

It seems that there is no easy way to browse a list of Authors, drilling down to see works under them. This is the main way I'd like to browse my collection.....

There is kind of a workaround which makes me think this should be relatively easy - when I look at a library, some of the entries are individual volumes, some are series, and some are authors. I can identify the Authors because they have an "image" icon
If I try to filter by Format = Image, it doesn't work, all entries are returned. But if I click on an image / author entry, then click on the image tag, I get taken back to the library view filtered by image, and this time the filter works, giving me the view I'm looking for. It would be nice if this view were easily accessible rather than requiring this workaround...

In addition

The list on the image / author page doesn't consolidate volumes into series - this is strange since everywhere else they're consolidated. If there is some reason why they can't be consolidated, it would be nice to at least sort by series

I use author here because I'm just testing with ebooks, but the same principle should ideally apply to all the filter fields, since presumably people would like to browse by editor, illustrator, etc. etc.

The image for the author could be editable rather than just picking a random work (nice to have)

![jjdXdiDcvkrMMJ4kNZw6n0YXqvJmpRn70nargHhTnX2Mlq86HJWd420Z3O0e5JZP-ss-3](https://github.com/Kareadita/Kavita/assets/30128380/4ba5f61e-cc3e-47de-bc4c-43b0b0ba2f05)

### Duration of Using Kavita

42

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by EmilGH** [comment]
^^^ This

**Comment by evanharmon1** [comment]
+1 I'm using for regular books and even with 100 books it's kind of hard to browse. E.g, I don't always remember what book I'm thinking of, but I know it's "that other book by Kafka that I haven't read..." etc.

**Comment by majora2007** [comment]
Update for all: I have partially implemented this for the v0.8.4. This is Artist/Writer only. We can expand it further once I get feedback from nightly users and stable (post v0.8.4). 

This took 80 hours to refactor how People were managed in the database, but I understand the need for this and that's why I put the work forward to deliver it. I hope with some extra feedback, we can expand upon this foundation and continue to build out the features needed by the community. 

https://github.com/Kareadita/Kavita/pull/3286

**Reply by bnerickson** [reply]
Thanks for implementing this, @majora2007 !

I think my only request is that the user be allowed to browse by Artist/Writer within their respective libraries individually as opposed to a global "Browse Authors" view.  Attempting to browse by author for ALL my libraries results in a lot of clutter unfortunately.  That isn't to say that the global view option should be removed (maybe made optional), but here's a silly mock-up screenshot of what I'm suggesting:

<img width="179" alt="browse_library_by_author" src="https://github.com/user-attachments/assets/86a5b0c3-4ba0-41fe-a131-6d3e31a26080">

Of course, selecting the "Browse Authors" subsection under the "Manga" library would bring up the same view as the global "Browse Authors" view except filtered so that only Manga authors are displayed.  The "Browse Genre" sub-section is just there to illustrate the kind of scalable "browse by..." options that users may want on a per-library basis in the future (see https://github.com/Kareadita/Kavita/discussions/3173); so designing/implementing this feature would probably do well to think at it from that perspective.  Let me know your thoughts, especially on whether I should break this out into its own separate feature request.  Thanks again for all your work on Kavita!

**Reply by majora2007** [reply]
Let's break this out into it's own section. I do think this is a good way to bring out more exploratory actions and keep it limited. The browse Authors/Artists is a first pass btw. I always do a first pass on the idea to get people's minds stimulated then count on their feedback to further explore the functionality.

**Reply by bnerickson** [reply]
https://github.com/Kareadita/Kavita/discussions/3445

Thanks again!

**Comment by majora2007** [comment]
This is fully implemented as part of the v0.8.7 release. In addition, you can filter by other items as well. 

https://github.com/Kareadita/Kavita/pull/3835
